rules/PHPUnit110/Rector/Class_/NamedArgumentForDataProviderRector.php

```diff
@@ -0,0 +1,291 @@
+                        {
+                            return [
+                                [
+                                    'keyA' => true,
+                                    'keyB' => false,
+                                ]
+                            ];
+                        }
+
+                        #[\PHPUnit\Framework\Attributes\DataProvider('dataProviderArray')]
+                        public function testFilter(bool $changeToKeyA, bool $changeToKeyB): void
+                        {
+                        }
+                    }
+                    CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+                    use PHPUnit\Framework\TestCase;
+
+                    final class SomeTest extends TestCase
+                    {
+                        public static function dataProviderArray(): array
+                        {
+                            return [
+                                [
+                                    'changeToKeyA' => true,
+                                    'changeToKeyB' => false,
+                                ]
+                            ];
+                        }
+
+                        #[\PHPUnit\Framework\Attributes\DataProvider('dataProviderArray')]
+                        public function testFilter(bool $changeToKeyA, bool $changeToKeyB): void
+                        {
+                        }
+                    }
+                    CODE_SAMPLE
+                    ,
+                ),
+            ],
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): Node|null
+    {
+        if (! $this->testsNodeAnalyzer->isInTestClass($node)) {
+            return null;
+        }
+
+        $wasChanged = false;
+        foreach ($node->getMethods() as $classMethod) {
+            if (! $classMethod->isPublic()) {
+                continue;
+            }
+
+            if ($classMethod->getParams() === []) {
+                continue;
+            }
+
+            $dataProviderMethodName = $this->getDataProviderMethodName($classMethod);
+
+            if ($dataProviderMethodName === null) {
+                continue;
+            }
+
+            $dataProviderMethod = $node->getMethod($dataProviderMethodName);
+            if ($dataProviderMethod === null) {
+                continue;
+            }
+
+            $namedArgumentsFromTestClass = $this->getNamedArguments($classMethod);
+
+            foreach ($this->extractDataProviderArrayItem($dataProviderMethod) as $dataProviderArrayItem) {
+                $wasChanged = $this->refactorArrayKey(
+                    $dataProviderArrayItem,
+                    $namedArgumentsFromTestClass
+                ) || $wasChanged;
```

Comment:
it can be more readable with:

```php
if ($this->refactorArrayKey(
                    $dataProviderArrayItem,
                    $namedArgumentsFromTestClass
                )) {
    $wasChanged = true;
}
```